src/main/java/nl/tudelft/planningstool/database/entities/User.java

```diff
@@ -72,7 +72,8 @@
     @OneToMany(mappedBy = "user")
     private Set<CourseRelation> courses = Sets.newHashSet();
 
-    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL)
+    // http://stackoverflow.com/questions/549961/hibernate-removing-item-from-a-list-does-not-persist#comment12057916_550441
+    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, orphanRemoval = true)
```

Comment:
:+1: This is indeed the way to map Objects that cannot live without a parent object. `CascadeType.ALL` only ensures all `UserOccurrences` belonging to a specific user are removed then a `User` is removed.